backend/src/libs/packages/controller/controller.package.ts

```diff
@@ -57,6 +58,8 @@ class Controller implements IController {
       fileToUpload,
       user,
       origin: headers.origin,
+      refererOrigin: processRefererHeader(headers.referer),
+      sharedArticleToken: headers['shared-article-token'] as string,
```

Comment:
Don't pass it to each request - take it only in place where you need it,, here you need only to pass `headers`
```suggestion

```